admin-dev/themes/new-theme/js/pages/product/edit/attachments-manager.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+import ProductEventMap from '@pages/product/product-event-map';
+import Router from '@components/router';
+import {createAttachment, getAttachmentInfo} from '@pages/product/services/attachments-service';
+
+const {$} = window;
+
+export default class AttachmentsManager {
+  constructor() {
+    this.$attachmentsContainer = $(ProductMap.attachments.attachmentsContainer);
+    this.$attachmentsCollection = $(ProductMap.attachments.attachmentsCollection);
+    this.$attachmentsTableBody = $(ProductMap.attachments.attachmentsTableBody);
+    this.prototypeTemplate = this.$attachmentsCollection.data('prototype');
+    this.prototypeName = this.$attachmentsCollection.data('prototypeName');
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.router = new Router();
+    this.init();
+  }
+
+  /**
+   * @private
+   */
+  init() {
```

Comment:
```suggestion
  init() {
```

Use typescript instead :)
ping @NeOMakinG

Reply:
The whole product page is not migrated yet, I think @jolelievre wanted to do it, but this should be done in another PR